src/prefect/utilities/executors.py

```diff
@@ -286,3 +286,56 @@ def run_with_ctx(*args: Any, _ctx_dict: dict, **kwargs: Any) -> Any:
         return fut.result(timeout=timeout)
     except FutureTimeout:
         raise TimeoutError("Execution timed out.")
+
+
+class RecursiveCall(Exception):
+    def __init__(self, func: Callable, *args: Any, **kwargs: Any):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+
+def tail_recursive(func: Callable) -> Callable:
```

Comment:
So I think we need to use `functools.wraps` here, because it appears that this decorator is hiding the `run` method of the runner classes from our API documentation (e.g., https://github.com/PrefectHQ/prefect/blob/167f8afd660e3f24225be168974d533383ee636f/src/prefect/engine/runner.py#L42)

Reply:
great catch, I'll incorporate